### PR TITLE
test: Regression tests for recent improvements and bug fixes

### PR DESCRIPTION
**Pull request by eriksvedang**

### CODE REVIEW

File: test/regression.carp [hunk 1]
```diff
@@ -51,6 +51,35 @@
...
+;; defining function with def (not defn)
+(defn duplicate-arg [f]
+  (fn [x] (f x x)))
+
+(def double (duplicate-arg +))
+
+(defn defining-function-with-def []
+  (double 3))
+
+;; using special symbol as binder name
```
Comment (hellerve) [inline test/regression.carp:63]: Is the test here just defining it? If so, maybe a mention of the issue number would be good!

File: test/regression.carp [hunk 2]
```diff
@@ -51,6 +51,35 @@
...
+(defndynamic special-symbol-as-binder []
+  (let [c (car (list 1 2 3))]
+     (list c)))
+
+;; dynamic closure referring to itself
```
Comment (hellerve) [inline test/regression.carp:68]: See above.

**Review by hellerve — APPROVED** [review]
Thank you!

File: test/regression.carp (hunk 2 above)
Comment (eriksvedang) [reply test/regression.carp:68]: Yes, will fix.

**Comment by scolsen** [comment]
Thanks!

**Pull request merged**